exercises/async-activity-completion/solution/workflow.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+from typing import List
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from datetime import timedelta
+
+class GreetingComposer:
+    def __init__(self, client: Client) -> None:
+        self.client = client
+
+    @activity.defn
+    async def compose_greeting(self) -> str:
+        print("Completing activity asynchronously")
+        asyncio.create_task(
+            self.complete_greeting(activity.info().task_token)
+        )
+        activity.raise_complete_async()
+
+    async def complete_greeting(self, task_token: bytes) -> None:
+        handle = self.client.get_async_activity_handle(task_token=task_token)
+        for _ in range(0, 3):
+            print("Waiting one second...")
+            await handle.heartbeat()
```

Comment:
This can fail with a `temporalio.client.AsyncActivityCancelledError` if the activity was since cancelled and the user is expected to report cancellation via the handle (but doesn't technically have to, server can still heartbeat timeout when this throws)